src/GraphODataTemplateWriter/Extensions/OdcmModelExtensions.cs

```diff
@@ -337,6 +337,62 @@ public static bool HasEnumProperties(this OdcmClass complex)
             return complex.Properties.Any(property => property.Type.AsOdcmEnum() != null);
         }
 
+        /// <summary>
+        /// Indicates whether the entity's base type is referenced as the type of another property.
+        /// TODO: remove filters on "entity" once we have annotation support.
+        /// </summary>
+        /// <param name="odcmClass">The OdcmClass to inspect.</param>
+        /// <returns>A value of true indicates that OdcmClass's base type is referenced as the type
+        /// of another property in a different entity or complex type.</returns>
+        public static bool IsBaseReferencedAsPropertyType(this OdcmClass odcmClass)
+        {
+            if (odcmClass.Base == null)
+            {
+                return false;
+            }
+            else
+            {
+                var isReferencedInAction = odcmClass.Namespace.Classes
+                    .SelectMany(c => c.Methods)  
+                    .Where( m => !m.IsFunction) // only get the Actions
+                    .Any(m => m.Parameters
+                        .Any( param => param.Type.Name == odcmClass.Base.Name 
+                                  && param.Type.Name != "entity"));
```

Comment:
```suggestion
                                  && "entity".Equals(param.Type.Name, StringComparison.OrdinalIgnoreCase)));
```
protects you from casing differences and the name being null